nasa-pictures-stealer/src/main/java/nasapicturesstealer/infrastructure/repository/NasaPicturesRepository.java

```diff
@@ -0,0 +1,187 @@
+package nasapicturesstealer.infrastructure.repository;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import nasapicturesstealer.domain.entity.NasaCamera;
+import nasapicturesstealer.domain.entity.NasaPicture;
+import nasapicturesstealer.infrastructure.entity.NasaCameraEntity;
+import nasapicturesstealer.infrastructure.entity.NasaPictureEntity;
+import nasapicturesstealer.infrastructure.mapper.NasaCameraMapper;
+import nasapicturesstealer.infrastructure.mapper.NasaPictureMapper;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class NasaPicturesRepository {
```

Comment:
@mvorotniak I like that you wrote so much code, but it seems like it's a bit overhead. The API we use structures photos by `sol` which means a solar day at Mars. So when you want to steam pictures by `sol` you should only check if you've done it before. If not, you store all the pictures:
- resolve the camera (find or create)
- same picture with a link to the camera

Therefore the overall logic would be much easier...

Reply:
I need this logic in case someone wants to add a picture for an existing camera, so a new duplicated camera is not created and the picture is assigned to the stored camera with id